standards/gitcoinGrants948.md

```diff
@@ -0,0 +1,127 @@
+# [WIP] ERC948 Subscription Standard (v1)
+
+## Summary
+The following is a proposed standard for subscription payments with read and write specs.
+
+## Abstract
+This standard allows a user to set a recurring payment to a service provider. This recurring transaction is ideally executed by a third party agent, but at a minimum the service provider themselves can execute the payment. Either side of the subscription should be able cancel the agreement with a reasonable amount of forewarning to the other party. This standard or similar should be integrated with future smart wallets to enable subscription payments in the Ethereum ecosystem.
+
+## Motivation
+Recurring payments are a widely use monetization method in traditional software. Currently, no such pattern exists with desirable UX traits in the decentralized space. Creating such a pattern will unlock many new avenues for revenue and provide an alternative to the dominant rent seeking and token-valuation based monetization methods in the current ecosystem.
+
+### ERC948 Read
+
+```
+interface ERC165 {
+
+@notice Query if a contract implements an interface
+@param interfaceID The interface identifier, as specified in ERC-165
+@dev Interface identification is specified in ERC-165. This function uses less than 30,000 gas.
+@return `true` if the contract implements `interfaceID` and `interfaceID` is not 0xffffffff, `false` otherwise
+
+function supportsInterface(bytes4 interfaceID) external view returns (bool);
+
+}
+
+interface ERC948Read is ERC165 {
+
+@dev Return the details of a subscription
+@param subscriptionId - A unique representation of the subscription based on the details of the agreement
+
+function getSubscription(
+   bytes subscriptionId
+ )
+ public
+ view
+ returns (
+   address _destination,
+   address _recipient,
+   address _agent,
+   uint _agentRewardPct,
```

Comment:
Is agent a processor (whoever is supposed to execute recurring payments)? I don't think this should be part of the standard - support for different business models (processors, fees, etc.) should come as an extension on top of the ERC948. Imo standard should support the simplest possible use case: recurring payments between a user and a provider (where provider takes care of recurring execution).

Reply:
Agent is a processor in the context of your implementation. Agent is placed here to allow for future implementations of third party processing. In this architecture, each user would be owner of their subscriptions contract. Thus, each user may choose a different agent or agent service for each subscription vs. having one processor to one provider as in your implementation. Agent may remain empty and recipient will always have the ability to pull down funds.

Reply:
I also think that processor-specific definitions are not the part of ERC-948, this can be handled externally, take a look at ERC-1228 as an example.